pkg/menu/menu.go

```diff
@@ -49,25 +70,60 @@ func processInput(input, warning *widgets.Paragraph, isValid func(string) (strin
 				warning.Text = ""
 				ui.Render(warning)
 			}
-			input.Text += e.ID
+			input.Text += k
 			ui.Render(input)
 		}
 	}
 }
 
-// GetInput will present an input box to user and return the user's input.
-// GetInput will check validation of input using isValid function.
-func GetInput(introwords string, location int, wid int, ht int, isValid func(string) (string, bool)) (string, error) {
-	intro := newParagraph(introwords, false, location, len(introwords)+4, 3)
-	location += 2
-	input := newParagraph("", true, location, wid, ht+2)
-	location += ht + 2
-	warning := newParagraph("", false, location, wid, 3)
+// create a new ui window and display an input box.
+func NewCustomInputWindow(introwords string, wid int, ht int, isValid func(string) (string, bool)) (string, error) {
+	uiEvents := ui.PollEvents()
+	return internalNewInputWindow(introwords, wid, ht, isValid, uiEvents)
+}
 
-	ui.Render(intro)
-	ui.Render(input)
-	ui.Render(warning)
+// open a new input window with fixed width=100, hight=1
+func NewInputWindow(introwords string, isValid func(string) (string, bool)) (string, error) {
+	uiEvents := ui.PollEvents()
+	return internalNewInputWindow(introwords, 100, 1, isValid, uiEvents)
+}
+
+func internalNewInputWindow(introwords string, wid int, ht int, isValid func(string) (string, bool), uiEvents <-chan ui.Event) (string, error) {
```

Comment:
this may seem weird, but it is common in go to just make the first letter lower case. So:
newInputWindow
would be fine here.